pysages/ssages/ffs.py

```diff
@@ -0,0 +1,181 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020: SSAGES Team (see LICENSE.md)
+
+from collections import namedtuple
+from jax import scipy
+from pysages.ssages.grids import build_indexer
+
+from .core import GriddedSamplingMethod, generalize  # pylint: disable=relative-beyond-top-level
+
+import jax.numpy as np
+
+# ================================== #
+#        Forward Flux Sampling       #
+# ================================== #
+
+class FFSState(
+    namedtuple(
+        "FFSState",
+        ("Phi_A", "Previous_Window", "Current_Window", "Prob_window"),
+    )
+):
+    def __repr__(self):
+        return repr("PySAGES " + type(self).__name__)
+
+#FFS needs a 1D grid in Collective Variable to define the path
+class FFS(GriddedSamplingMethod):
+    def __call__(self, snapshot, helpers):
+        #Number of parallel replicas per window
+        Nmax_replicas= np.asarray(self.kwargs.get('M', 2000))
+        N0_steps= np.asarray(self.kwargs.get('M', 2000))
+        sampling_time= np.asarray(self.kwargs.get('M', 2000))
+        system=0
+        run=0
+        return _ffs(snapshot, self.cv, self.grid, Nmax_replicas, N0_steps,sampling_time,system,run,helpers)
+
+
+def _ffs(snapshot, cv, grid, Nmax_replicas, N0_steps,sampling_time,system,run,helpers):
+    dt = snapshot.dt
+    dims = grid.shape.size
+    N_windows=len(grid)
+    indices, momenta = helpers.indices, helpers.momenta
+#initialize method
+    def initialize():
+        Phi_A = 0.0
+        Previous_Window = 0.0
+        Current_Window =0.0
+        Prob_window= 0.0
+        return FFSState(Phi_A,Previous_Window,Current_Window,Prob_window)
+ #Function to determine if the initial snapshot is good (below first window)   
+    def good_input(snapshot,cv,grid,rs,ids):
+        ξ, Jξ=cv(rs, indices(ids))
+        if ξ<grid[0]:
+            return True
+        else:
+            return False
+#Function to generate initial configurations from the Basin A
+#requires system to integrate, how many snapshots you need, the window-A threshold and sampling time 
+#The function returns a list of snapshots
```

Comment:
Maybe we can convert these comments into a docstring. Same for all comments that describe what each method does.